src/_base/helm/app/templates/service/elasticsearch/deployment.yaml

```diff
@@ -1,25 +1,26 @@
-{{ if .Values.service.elasticsearch | default .Values.docker.services.elasticsearch.enabled }}
+{{- with .Values.docker.services.elasticsearch -}}
+{{- if $.Values.service.elasticsearch | default .enabled -}}
 apiVersion: apps/v1
 kind: Deployment
 metadata:
-  name: {{ .Values.resourcePrefix }}elasticsearch
+  name: {{ $.Values.resourcePrefix }}elasticsearch
   annotations:
     argocd.argoproj.io/sync-wave: "4"
   labels:
-    app.service: {{ .Values.resourcePrefix }}elasticsearch
+    app.service: {{ $.Values.resourcePrefix }}elasticsearch
 spec:
   replicas: 1
   selector:
     matchLabels:
-      app.service: {{ .Values.resourcePrefix }}elasticsearch
-{{- if .Values.persistence.elasticsearch.enabled }}
+      app.service: {{ $.Values.resourcePrefix }}elasticsearch
+{{- if .enabled }}
```

Comment:
Missing persistence check? Though I suppose if using host mounts, two elasticsearch instances on the same node for rolling update is bad too.

Reply:
The same problem for two instances accessing same hostmount apply to all the services we Recreate instead of rolling update

Reply:
the logic is saying if no persistence then no need for Recreate strategy, and Recreate needed if persistence so volume isn't in use by two pods the same time

Reply:
emptyDir volumes aren't actually shared volumes